### PR TITLE
Add color chips to the color scheme dropdown in Appearance

### PR DESCRIPTION
**Pull request by PankajBhojwani**
Does what it says on the tin.

### CODE REVIEW

**Comment by PankajBhojwani** [comment]
chipDropdown

**Comment by carlos-zamora** [comment]
Since the XAML is very similar to that of the color schemes page, is there any way we can deduplicate the code? Maybe introduce something similar to `SettingContainer` (i.e. `ColorSchemeComboBoxItem`) where it takes in a `ColorSchemeViewModel` and presents it in such a way?

**Comment by zadjii-msft** [comment]
oh it's so pretty

**Comment by PankajBhojwani** [comment]
> Since the XAML is very similar to that of the color schemes page, is there any way we can deduplicate the code? Maybe introduce something similar to SettingContainer (i.e. ColorSchemeComboBoxItem) where it takes in a ColorSchemeViewModel and presents it in such a way?

Since the values are slightly different between the two UIs (namely the sizes/spacings), we will need 2 separate styles, one for the color schemes list view and one for the color schemes combo box.... This means we are simply moving the code duplication from one place to another.

File: src/cascadia/TerminalSettingsEditor/Appearances.xaml [hunk 1]
```diff
@@ -49,7 +57,120 @@
...
+                                                Grid.Column="6"
+                                                Content="{x:Bind ColorEntryAt(14), Mode=OneWay}"
+                                                ContentTemplate="{StaticResource ColorChipTemplate}"
+                                                IsTabStop="False" />
+                                <ContentControl Grid.Row="1"
+                                                Grid.Column="7"
+                                                Content="{x:Bind ColorEntryAt(15), Mode=OneWay}"
+                                                ContentTemplate="{StaticResource ColorChipTemplate}"
+                                                IsTabStop="False" />
+                                <TextBlock Grid.RowSpan="2"
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/Appearances.xaml:163]: The only suggestion I have is to add a `TextTrimming` here. Personally a fan of `WordEllipsis`

https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.textblock.texttrimming?view=winrt-22621#windows-ui-xaml-controls-textblock-texttrimming

**Review by carlos-zamora — APPROVED** [review]
Eh, the deduplication of code seems like a decent amount of extra work for little gain. I don't even know if it's worth filing a follow-up. If anybody else agrees, sure. But I don't think it's worth blocking over.

Excited to see this merge!

**Comment by DHowett** [comment]
Wow, that other PR made this one super easy

**Review by zadjii-msft — APPROVED** [review]

**Comment by carlos-zamora** [comment]
Bug Bash Feedback:
- DH: I feel like the padding between the list cell and the background could be smaller and more even on all sides

**Comment by PankajBhojwani** [comment]
> DH: I feel like the padding between the list cell and the background could be smaller and more even on all sides

We can change the `Padding` property of the `ComboBox` and make the spacing smaller when the combo box is unexpanded, like so:

image

However, for some reason this does not translate over to when the `ComboBox` is expanded, I have a feeling xaml is overriding us for some reason

image

File: src/cascadia/TerminalSettingsEditor/Appearances.xaml (hunk 1 above)
Comment (PankajBhojwani) [reply src/cascadia/TerminalSettingsEditor/Appearances.xaml:163]: Added that! Thanks

File: src/cascadia/TerminalSettingsEditor/Appearances.xaml [hunk 2]
```diff
@@ -54,6 +54,7 @@
                                     SettingOverrideSource="{x:Bind Appearance.DarkColorSchemeNameOverrideSource, Mode=OneWay}">
                 <ComboBox ItemsSource="{x:Bind Appearance.SchemesList, Mode=OneWay}"
                           SelectedItem="{x:Bind Appearance.CurrentColorScheme, Mode=TwoWay}"
+                          Padding="4"
                           Style="{StaticResource ComboBoxSettingStyle}">
                     <ComboBox.ItemTemplate>
                         <DataTemplate x:DataType="local:ColorSchemeViewModel">
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/Appearances.xaml:60]: i was thinking more like, we could reduce the inner padding inside the grid to generally tighten it up :)

Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/Appearances.xaml:60]: nah, it looks fine. let's ship it.

**Review by DHowett — APPROVED** [review]

**Pull request merged**